src/Filter/AbstractDateFilter.php

```diff
@@ -49,43 +50,47 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
         $data['type'] = !isset($data['type']) || !is_numeric($data['type']) ? DateType::TYPE_EQUAL : $data['type'];
 
         // Some types do not require a value to be set (NULL, NOT NULL).
-        if (!$this->typeRequiresValue($data['type']) && !$data['value']) {
+        if ($this->typeDoesRequireValue($data['type']) && !isset($data['value'])) {
```

Comment:
We could perform the cheaper check before.
```suggestion
        if (!isset($data['value']) && $this->typeDoesRequireValue($data['type'])) {
```